crates/shared/src/bad_token/token_owner_finder/seasolver.rs

```diff
@@ -0,0 +1,45 @@
+use super::TokenOwnerProposing;
+use anyhow::Result;
+use ethcontract::H160;
+
+pub struct SeasolverTokenOwnerFinder {
+    owners: Vec<H160>,
+}
+
+impl SeasolverTokenOwnerFinder {
+    pub fn new(owners: Vec<H160>) -> Self {
+        Self { owners }
+    }
+}
+
+#[async_trait::async_trait]
+impl TokenOwnerProposing for SeasolverTokenOwnerFinder {
+    async fn find_candidate_owners(&self, _: H160) -> Result<Vec<H160>> {
+        Ok(self.owners.clone())
+    }
```

Comment:
If I understood @nlordell's recent changes for the trade simulation stuff correctly we could reuse the idea to do bad token detection without relying on finding token owners anymore by just using state overrides instead, right?

Reply:
Correct. This requires the `roundtripping` implementation to in the `Trader.sol` to be complete.

Basically, we need to port https://github.com/nlordell/cow-trade-simulation/blob/42222a7b422f5f49b7f8b16c0c2df744bb38603d/contracts/Trader.sol#L87 to this project.